lib/src/signed_video_h26x_auth.c

```diff
@@ -907,13 +941,36 @@ maybe_validate_gop(signed_video_t *self, h26x_nalu_t *nalu)
     // If this is the first arrived SEI, but could still not validate the authenticity, signal to
     // the user that the Signed Video feature has been detected.
     if (validation_flags->is_first_sei) {
+      // Only set authenticity to SV_AUTH_RESULT_SIGNATURE_PRESENT if it has not been marked as
+      // NOT_OK This ensures that an earlier failure is not overwritten.
       latest->authenticity = SV_AUTH_RESULT_SIGNATURE_PRESENT;
```

Comment:
Your latest change adds no value. In fact it will mark latest to "signature present" even if it is not. What I meant was that you should **replace** the line and not **complement** it.